docs/content/TopNQuery.md

```diff
@@ -211,4 +211,7 @@ Users who can tolerate *approximate rank* topN over a dimension with greater tha
     "queryType": "topN",
     "threshold": 2
 }
-```
\ No newline at end of file
+```
+
+### DimExtractionFn
+A TopN over a dimension which is extracted has its own code path to be able to properly capture re-bucketing data. If the query issuer knows that data does not need to be rebucketed (for example, changing a name from one unique value to another unique value), a query context of `topNFastRename` set to `true` can be specified to enable a more optimized codepath.
```

Comment:
Given that I think that you have setup the optimization for joining on the broker side now, do we need the context?  It should be able to just re-write the query and send it down, no?